x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportCreateTrainedModelAllocationAction.java

```diff
@@ -0,0 +1,82 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.core.ml.action.CreateTrainedModelAllocationAction;
+import org.elasticsearch.xpack.core.ml.action.CreateTrainedModelAllocationAction.Request;
+import org.elasticsearch.xpack.core.ml.action.CreateTrainedModelAllocationAction.Response;
+import org.elasticsearch.xpack.ml.inference.allocation.TrainedModelAllocationClusterService;
+import org.elasticsearch.xpack.ml.inference.allocation.TrainedModelAllocationNodeService;
+import org.elasticsearch.xpack.ml.inference.allocation.TrainedModelAllocationService;
+import org.elasticsearch.xpack.ml.inference.deployment.DeploymentManager;
+
+public class TransportCreateTrainedModelAllocationAction extends TransportMasterNodeAction<Request, Response> {
+
+    private final TrainedModelAllocationClusterService trainedModelAllocationClusterService;
+
+    @Inject
+    public TransportCreateTrainedModelAllocationAction(
+        TrainedModelAllocationClusterService trainedModelAllocationClusterService,
+        TrainedModelAllocationService trainedModelAllocationService,
+        DeploymentManager deploymentManager,
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver
+    ) {
+        super(
+            CreateTrainedModelAllocationAction.NAME,
+            false,
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            Request::new,
+            indexNameExpressionResolver,
+            Response::new,
+            ThreadPool.Names.SAME
+        );
+        this.trainedModelAllocationClusterService = trainedModelAllocationClusterService;
+        // Here we create our singleton for the node service
+        clusterService.addListener(
+            new TrainedModelAllocationNodeService(
+                trainedModelAllocationService,
+                clusterService,
+                deploymentManager,
+                transportService.getTaskManager(),
+                threadPool
+            )
```

Comment:
This is pretty much what persistent tasks do. Its ok as transport classes are singletons and are created on every node where the plugin is loaded.